### PR TITLE
Update NTP with async image load

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves : https://github.com/brave/brave-browser/issues/8892
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            if (mWallpaper.getLogoPath() != null ) {
+                try {
+                    Uri logoFileUri = Uri.parse("file://"+ mWallpaper.getLogoPath());
+                    InputStream inputStream = mContext.getContentResolver().openInputStream(logoFileUri);
+                    logoBitmap = BitmapFactory.decodeStream(inputStream);
+                } catch(FileNotFoundException exc) {
+                    Log.e("NTP", exc.getMessage());
+                }
+            }
+        }
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java:58]: nit: I would make it as a separate method, but it is not critical and is up to author

File: android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Override
+    protected Pair<Bitmap, Bitmap> doInBackground() {
+        Bitmap logoBitmap = null;
+        if (mNTPImage instanceof NTPBackgroundImagesBridge.Wallpaper) {
+            NTPBackgroundImagesBridge.Wallpaper mWallpaper = (NTPBackgroundImagesBridge.Wallpaper) mNTPImage;
+            if (mWallpaper.getLogoPath() != null ) {
+                try {
+                    Uri logoFileUri = Uri.parse("file://"+ mWallpaper.getLogoPath());
+                    InputStream inputStream = mContext.getContentResolver().openInputStream(logoFileUri);
```
Comment (AlexeyBarabash) [inline android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java:52]: We have to close the stream below in finally block.
There are messages ` StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at attached stack trace but never released.` pointing to this place.


The same for https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/ntp_background_images/NTPUtil.java#L261 which is not a part of this PR.

**Comment by AlexeyBarabash** [comment]
Looks like this import is not required anymore
https://github.com/brave/brave-core/pull/5087/files#diff-95038ab914b43b26a5b638ec136805afR22

File: android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java (hunk 1 above)
Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java:58]: @AlexeyBarabash i am updating it custom homepage as we have many things to add here. i have a separate async just to get bitmap.

File: android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java (hunk 2 above)
Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/ntp_background_images/FetchWallpaperWorkerTask.java:52]: @AlexeyBarabash Good point. sorry i missed it. updating it.

**Comment by kjozwiak** [comment]
Reproduced the issue using `Samsung Tab A` running `Android 9` using the following build:

```
Brave 1.5.131 CR: 80.0.3987.162
```

Screenshot_20200424-171119_Brave

Screenshot_20200424-171148_Brave

Verification **PASSED**  on Samsung Tab A` running `Android 9` using the following build:

```
Brave - Nightly 1.10.9 CR: 81.0.4044.122
```

**`Case #1`**

* Open a NTP
* Open another tab and load https://www.google.ca and rotate into `landscape`
* switch back to the NTP (you'll notice it being cut off)

**`Case #2`**

* Open a NTP
* Open another tab and load https://www.google.ca
* switch back to the NTP and rotate to `landscape`
* switch back to the tab with https://www.google.ca and rotate to `portrait`
* switch back to the NTP

**Pull request merged**